### DISCUSSION TITLE
Prompt customization doesn't seem to be supported

### DISCUSSION THREAD

**Discussion by dj-256** [Microsoft/node-pty#620] [Q&A] [answered]
## Environment details

- OS: ubuntu (docker image)
- OS version: 22.04
- node-pty version: 1.0.0

## Issue description
When using a custom prompt definition - in my case `echo "PS1=\"\e[1;32m\u@tuto\e[0m:\e[1;34m\w\e[0m\$ \"" >> /root/.bashrc`, node-pty does not seem to understand it and some weird things happen on the UI. The first line of the terminal is reduced and gets rewritten when it's exceeded. Stranger things happen when trying to erase text in the terminal. I'm using xterm.js with sockets.

**Comment by Tyriar** [comment]
> The first line of the terminal is reduced and gets rewritten when it's exceeded.

It sounds like you aren't syncing the front and backend's dimensions https://github.com/xtermjs/xterm.js/wiki/FAQ#characters-are-being-overwritten-when-wrapped-to-a-new-line

**Reply by dj-256** [reply]
Hi, thank you for the quick answer. I did sync the rows and columns of the terminal. Are there any other things to sync?

**Reply by dj-256** [reply]
I've tried creating a reproduction repo to better show the issue I'm facing : https://github.com/dj-256/node-pty-custom-prompt

**Reply by Tyriar** [reply, marked as answer]
After a quick skim of the code that does look correct. It might be an issue with your shell or something as this definitely does work or we'd hear about it not working in VS Code. My guess is you need to tell bash which parts of the prompt are formatting via `\[` and `\]`.

**Reply by dj-256** [reply]
Ok thanks, this was indeed the issue. I needed to wrap the parts that were formatting with `\[` `\]`. Like so: `echo "PS1=\"\[\e[1;32m\]\u@tuto\[\e[0m:\e[1;34m\w\e[0m\]\$ \"" >> /root/.bashrc`
